.github/workflows/rust.yml

```diff
@@ -11,31 +11,97 @@ env:
   SQLX_OFFLINE: "true"
 
 jobs:
-  unit-tests:
-    runs-on: ubuntu-latest
+  unit-tests-darwin:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        arch: [x86_64, aarch64]
     steps:
       - name: Ensure musl installed
-        run: sudo apt-get install -y musl-dev musl-tools musl && rustup target add x86_64-unknown-linux-musl
+        run: rustup target add ${{ matrix.arch }}-apple-darwin
 
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
       - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84 # v2
+          
+      - name: Run unit tests
+        run: cargo test --target ${{ matrix.arch }}-apple-darwin
+
+  unit-tests-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [x86_64, aarch64]
+    steps:
+      - name: Ensure toolchain installed
+        run: sudo apt-get install -y musl-dev musl-tools musl gcc-aarch64-linux-gnu llvm clang qemu-user && rustup target add ${{ matrix.arch }}-unknown-linux-musl
+
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84 # v2
+
+      - name: Set up environment variables
+        run: |
+          if [ "${{ matrix.arch }}" == "aarch64" ]; then
+            echo "CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_RUSTFLAGS=-Clink-self-contained=yes -Clinker=rust-lld" >> $GITHUB_ENV
```

Comment:
This is required due to ring dependency

https://github.com/briansmith/ring/issues/1414

I'm not using clang as suggested by someone in the mr. Reused what exists in the ring build scripts to get musl to work on aarch64

Reply:
<img width="544" alt="Screenshot 2024-05-23 at 18 50 10" src="https://github.com/Isawan/iam-assumer/assets/18237750/6e575bd4-86e9-4ce0-b7dd-7e583bceef4c">